File: 2006-count-number-of-pairs-with-absolute-difference-k/2006-count-number-of-pairs-with-absolute-difference-k.js
/**
 * @param {number[]} nums
 * @param {number} k
 * @return {number}
 */
var countKDifference = function(nums, k) {
  const l = nums.length
  let count = 0
  for (let i = 0; i < l - 1; i++) {
    for (let j = i + 1; j < l; j++) {
      const delta = nums[i] - nums[j]
      if (delta === k || delta === -k) count++
    }
  }
  return count
};

File: 2006-count-number-of-pairs-with-absolute-difference-k/2006-count-number-of-pairs-with-absolute-difference-k.go
func countKDifference(nums []int, k int) int {
  if len(nums) <= 1 {
    return 0
  }
  ans := 0
  gt, le := nums[0] + k, nums[0] - k
  for i := 1; i < len(nums); i++ {
    if nums[i] == gt || nums[i] == le {
      ans ++
    }
  }
  
  return ans + countKDifference(nums[1:], k)
}